### PR TITLE
Remove `common.http`

### PR DESCRIPTION
**Pull request by mtuchi**
## Summary

Version lock `@openfn/language-common` to all adaptors that were using `workspace:^`. And removed `http` in common.

Ref #315 

## Review Checklist

Before merging, the reviewer should check the following items:

- [x] Does the PR do what it claims to do?
- [ ] Are there any unit tests? Should there be?
- [x] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.

### CODE REVIEW

**Comment by josephjclark** [comment]
I'm not sure this is right :thinking:  An adaptor with a common dependency like `^1.8.1` will not auto-update to common 2.0. It'll use the latest compatible version, which means the highest minor.

The only dependencies we need to worry about are `workspace:*` or `*` or something like that.

All we're doing here is restricting adaptors to not use patch or minor bumps of common - which could be important.

File: packages/bigquery/package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
   ],
   "dependencies": {
     "@google-cloud/bigquery": "^5.12.0",
-    "@openfn/language-common": "workspace:1.10.3",
+    "@openfn/language-common": "1.11.1",
```
Comment (josephjclark) [inline packages/bigquery/package.json:24]: I this case bigquery is locked to a specific version number. I don't know why (maybe we can do some github archaeology and work it out). 

This PR is updating the dependency to the next minor version. That's _probably_ fine but I wouldn't be comfortable doing it without checking first.

File: .changeset/lucky-pianos-sing.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+- Added version locking to adaptors that uses `openfn/language-common`
+  dependency to ensure consistent and predictable behavior across different
+  environments.
+- Removed `http` on `common` to improve compatibility and reduce potential
+  security vulnerabilities.
+
+### Deprecated
+
+- Deprecated the use of _http.method()_ Eg: `http.post()` as it is no longer a
```
Comment (josephjclark) [inline .changeset/lucky-pianos-sing.md:15]: `http.method()` isn't deprecated - it's removed entirely!

File: .changeset/lucky-pianos-sing.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  part of the `@openfn/language-common`.
+
+### Known Issues
```
Comment (josephjclark) [inline .changeset/lucky-pianos-sing.md:18]: We don't need to include a known issues section if there are no known issues :) To be honest I would prefer not to release software if there are known issues in it...

File: packages/common/test/http.test.js [hunk 4]
```diff
@@ -1,281 +0,0 @@
-import nock from 'nock';
```
Comment (josephjclark) [inline packages/common/test/http.test.js:1]: Are we still using `nock`? We should have moved entirely to the `undici` pattern by now

File: packages/common/src/http.js [hunk 5]
```diff
@@ -1,202 +0,0 @@
-import { expandReferences, splitKeys } from './Adaptor';
-import axios from 'axios';
```
Comment (josephjclark) [inline packages/common/src/http.js:2]: Have we removed the axios dependency?

File: packages/bigquery/package.json (hunk 1 above)
Comment (mtuchi) [reply packages/bigquery/package.json:24]: @josephjclark i am getting this error when i run `pnpm test`  if i use the workspace version 
```
> @openfn/language-bigquery@2.0.2 test /Users/openfn/Workspace/adaptors/packages/bigquery
> mocha --experimental-specifier-resolution=node --no-warnings


file:///Users/openfn/Workspace/adaptors/packages/bigquery/src/Adaptor.js:130
  http,
  ^^^^
SyntaxError: The requested module '@openfn/language-common' does not provide an export named 'http'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5)
```

Comment (mtuchi) [reply packages/bigquery/package.json:24]: That's why i locked the `common` version to `1.11.1` which is the latest version. Should i perhaps use `^1.10.3`

File: packages/common/test/http.test.js (hunk 4 above)
Comment (mtuchi) [reply packages/common/test/http.test.js:1]: Removed

File: packages/common/src/http.js (hunk 5 above)
Comment (mtuchi) [reply packages/common/src/http.js:2]: Removed

File: packages/bigquery/package.json (hunk 1 above)
Comment (josephjclark) [reply packages/bigquery/package.json]: That's actually quite an interesting problem. The workspace dependency, the actual adjacent source code, is no longer compatible with some modules.

We haven't yet bumped common and http to 2.0 have we? If we bump  them, we can make sure the dependencies are of the form `^1.10.3`. I don't think we need `workspace:` in front.

I wonder what `workspace:^1...` even means. Will it build a dependency from a tagged source branch? That' some serious muscle under the hood. I suspect we don't want `workspace:` and, strange as it seems, modules with legacy dependencies should download common from npm

**Comment by mtuchi** [comment]
Closing this in favor of #427

**Pull request closed without merging**